lib/bolt/target.rb

```diff
@@ -80,6 +80,22 @@ def to_h
       )
     end
 
+    def detail
+      {
+        'name' => name,
+        'uri' => uri,
+        'alias' => @target_alias,
```

Comment:
I think we should actually remove this if it's not set. Right now it's `nil` if it's not set, which results in:
```
...
alias:
config:
  transport: ssh
...
```
Which raises this error: `Alias entry on <hostname> must be a String or Array, not NilClass for group at ["all"]`. If we set it to an empty string if nil, then we get `Invalid alias for group at ["all"]`. We could theoretically set it to the name if it's unset, but that seems unnecessarily verbose. I think the best thing to do is to set `alias` in this hash after creating it if it has a value, otherwise omit it.

Reply:
Updated to replace all `nil` values with a stringified `"nil"`